code/modules/mob/living/brain/posibrain.dm

```diff
@@ -1,5 +1,48 @@
 GLOBAL_VAR(posibrain_notify_cooldown)
 
+GLOBAL_LIST_EMPTY(on_station_posis)
+
+/datum/job/cyborg/posibrain
+	title = JOB_NAME_POSIBRAIN
+	total_positions = 0
+	spawn_positions = 0
+	supervisors = "your laws" //No AI yet as you are just a cube
+	show_in_prefs = FALSE //No reason to show in preferences
+
+/datum/job/cyborg/posibrain/after_spawn_silicon(mob/living/silicon/robot/R, mob/M)
+	var/obj/item/mmi/posibrain/P = pick(GLOB.on_station_posis)
+
+	//Never show number of current posis
+	current_positions = 0
+
+	if(!P.activate(R)) //If we failed to activate a posi, kick them back to the lobby.
+		//Code taken from "send to lobby" admin panel
+		var/mob/dead/new_player/NP = new()
+		NP.ckey = M.ckey
+
+		to_chat(NP, "<span class='warning'>Failed to Late Join as a Posibrain. Look higher in chat for the reason.</span>")
+
+	qdel(R)
```

Comment:
Instead of picking a random one, loop through all of them until you find a valid one and then return when P.activate passes successfully.

Reply:
90 percent of failures will be because the user is either: 
1. Banned from Silicon Roles
2. Too new
3. 1 Posi per round (which shouldn't happen if they are in lobby)
4. Ghost spawns limited (very rare)
5. Suicided (shouldn't happen if they are in lobby)

The only one that will be different is if at the exact instance they click the join button, an in round ghost clicks "Yes" on the posi prompt (which you don't get joining from lobby), and there is only one posi/the random chance matches.

I'd have to refactor activate to return error codes on failure to kick out early, instead of spamming the chat for every posibrain that it tries. Do you want me to do that?

Reply:
You just have to return when it is successful, if it throws an error message in the chat then you would have hit a bugged state anyway.

Reply:
> You just have to return when it is successful, if it throws an error message in the chat then you would have hit a bugged state anyway.

Activate returns a message in chat telling you why you did not spawn as a posibrain. The errors are anywhere from too new to banned to someone already grabbed it.

If I loop over all of the available posibrains, and someone is banned from posibrain, it will say:
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"
"Banned from Posibrain"

"Failed to spawn, check above for reason"

This happens because that chat message exists in activate, where it needs to be so that ghost activations still work properly.

So, I'd have to add a return code that is more specific than "failure" so the loop can terminate early.